main.tf

```diff
@@ -1,3 +1,132 @@
-/********************************************************************
-This file is used to implement the ROOT module.
-*********************************************************************/
+###########################################################
+# This file creates an event notificaiton resource instance
+###########################################################
+locals {
+  # tflint-ignore: terraform_unused_declarations
+  validate_kms_plan = var.kms_encryption_enabled && var.plan != "standard" ? tobool("kms encryption is only supported for standard plan") : true
+  # tflint-ignore: terraform_unused_declarations
+  validate_kms_values = !var.kms_encryption_enabled && (var.kms_key_crn != null || var.root_key_id != null) ? tobool("When passing values for var.kms_key_crn or/and var.root_key_id, you must set var.kms_encryption_enabled to true. Otherwise unset them to use default encryption") : true
+  # tflint-ignore: terraform_unused_declarations
+  validate_kms_vars = var.kms_encryption_enabled && (var.kms_key_crn == null || var.root_key_id == null) ? tobool("When setting var.kms_encryption_enabled to true, a value must be passed for var.kms_key_crn and var.root_key_id") : true
+  # tflint-ignore: terraform_unused_declarations
+  validate_auth_policy = var.kms_encryption_enabled && var.skip_iam_authorization_policy == false && var.existing_kms_instance_guid == null ? tobool("When var.skip_iam_authorization_policy is set to false, and var.kms_encryption_enabled to true, a value must be passed for var.existing_kms_instance_guid in order to create the auth policy.") : true
+
+  # Determine what KMS service is being used for encryption
+  kms_service = var.kms_key_crn != null ? (
```

Comment:
This will need to be updated to extract the KMS service from the value of `existing_kms_instance_crn`

Reply:
Done.